### PR TITLE
add atom support

### PR DESCRIPTION
**Pull request by Apres2707**
added support for Atom transmitter

### CODE REVIEW

**Comment by tolot27** [comment]
@Apres2707 Thanks for your contribution. Would you please explain in more detail what the PR adds and if it addresses an existing issue? Also, please provide links and information about the Atom transmitter.

**Comment by Apres2707** [comment]
> @Apres2707 Thanks for your contribution. Would you please explain in more detail what the PR adds and if it addresses an existing issue? Also, please provide links and information about the Atom transmitter.

Hi
Atom - this is new transmitter from Michel Liu for freestyle libre sensor (with photos in the attachment). It is different from mm in appearance. Double-sided tape didn't required. The material is medical and waterproof.​ We hope to bring more convenience to patients through xdrip.
This PR adds the ability to connect xDrip+ to this transmitter.
FGkCdmeZOZE

**Comment by Navid200** [comment]
@Apres2707 Is there a web site for the manufacturer?

**Comment by Apres2707** [comment]
> @Apres2707 Is there a web site for the manufacturer?

Atom manufacturer don't have a website yet. But company already exists.

Yishi Technology (Shenzhen) Co., Ltd.
829B5, Building 3, Building 2, Dachong Business Center (Phase I), 9676 Shennan Avenue, Nanshan District, Shenzhen

**Comment by tolot27** [comment]
@Apres2707 Are you an employee of that company or how are you related to this product? Will you be the long-term responsible developer for the Atom xDrip integration?

We need at least some specification/documentation.

**Comment by jamorham** [comment]
@Apres2707 can we get a sample device to test with?

@tzachi-dar are there any implications with the changes to `LibreBlock` ?

**Comment by Apres2707** [comment]
> @Apres2707 Are you an employee of that company or how are you related to this product? Will you be the long-term responsible developer for the Atom xDrip integration?
> 
> We need at least some specification/documentation.
I am a company employee. What specific documents would you like?

> @Apres2707 can we get a sample device to test with?
> 
We are of course ready to provide you with an atom for testing. Need your address where to send the atom

**Comment by Navid200** [comment]
> We are of course ready to provide you with an atom for testing. Need your address where to send the atom

Please wait for jamorham to communicate with you, in case he hasn't already.

**Comment by jamorham** [comment]
@Apres2707 Please contact me via email jamorham at gmail and I will get you address details for test units thanks

**Comment by tzachi-dar** [comment]
The changes to libreblock are fine. I'll give a better look tomorrow.

**Comment by Navid200** [comment]
@Apres2707 Can I ask you a few questions?  If you feel the questions are unreasonable, obviously, you don't have to answer.

I am wondering what this new device offers users that the existing products lack.  You already mentioned that it does not require double sided tape.  Is that necessary for mm?
Is there any other advantage it offers?

Where is it available for the users to buy?  
For example, if I wanted to buy one in Canada, can I?
How about in Europe?
How about in the US?
How about in China?

Where is it manufactured?
What are the plans for productions?  How long will it continue?
Are there future roadmaps for other products or enhancements to existing products?

Are there any other comparisons you can provide to other existing products?
For example, battery run time, connectivity, ...

Are there any disadvantages compared to existing products?

When we add support to xDrip, it is not easy for us to discontinue providing support.  This, sometimes, could cause problems with supporting other devices.  For example, when we make a change to xDrip to address an issue with a product, we have to worry about how the change may negatively impact all the other devices xDrip supports.  
So, we need to make sure there is a good enough reason to support something before we engage.

Can you think of any other information you can provide that can help us?

Please note that I am one of the individuals who helps a little bit.  I have no authority here and cannot make any decisions and have no ownership of any part of this repository.  I am asking these questions only on my accord (not authorized by anyone) and as I said, you don't have to answer.

File: app/src/main/java/com/eveningoutpost/dexdrip/Models/Atom.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            if (s_full_data == null) {
+                InitBuffer(expectedSize);
+            }
+            addData(buffer);
+            return reply;
+
+        }
+
+        if (first == 0xBF) {
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```
Comment (tzachi-dar) [inline app/src/main/java/com/eveningoutpost/dexdrip/Models/Atom.java:115]: sdf is not used anywhere.

**Comment by tzachi-dar** [comment]
I have looked on this pr and it looks good.

**Comment by Apres2707** [comment]
@Navid200 
The use of double-sided tape is convenient for many patients. Before starting the development of a new transmitter, we conducted a survey among patients with diabetes mellitus. This was one of their wishes. Thus, the atom can be easily put on and removed from the sensor. We have already launched sales in China and Russia. You can order an atom from us without any problems. Worldwide delivery. And in the context of the coronavirus epidemic, delivery is becoming an urgent problem. We are now trying to fix problems that have arisen during trial sales. We are working on a more capacious battery.

File: app/src/main/java/com/eveningoutpost/dexdrip/Models/Atom.java (hunk 1 above)
Comment (Apres2707) [reply app/src/main/java/com/eveningoutpost/dexdrip/Models/Atom.java]: Corrected. Thank you for your comment

**Comment by tolot27** [comment]
@Apres2707 Please remove all the translated `strings-xx.xml` files and keep only `strings.xml`. Translation is done at [Crowdin](https://crowdin.com/project/xdrip) and untranslated strings will be displayed in English. That avoids merge conflicts with Crowdin commits.

Also, we still don't have any documentation and links to resources, yet.

**Comment by jamorham** [comment]
Testing is starting now on this with the supplied units.

File: wear/src/main/java/com/eveningoutpost/dexdrip/Models/Atom.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                if (buffer.length >= 11) {
+                    patchInfo = Arrays.copyOfRange(buffer, 5, 11);
+                }
+            }
+            return reply;
+        }
+        if (first == 0x82) {
+            int expectedSize = lens + ATOM_FOOTER;
+            if (s_full_data == null) {
+                InitBuffer(expectedSize)
```
Comment (jamorham) [inline wear/src/main/java/com/eveningoutpost/dexdrip/Models/Atom.java:101]: missing `;` here by the way

Comment (Apres2707) [reply wear/src/main/java/com/eveningoutpost/dexdrip/Models/Atom.java]: Thank you for your comment and I ask you to excuse me for my mistake. Fixed

**Comment by Apres2707** [comment]
> Testing is starting now on this with the supplied units.

Is it possible to find out the approximate time frame when some information about the test results may appear?
I am asking this question solely for understanding, if you do not answer, I will not insist, of course.

**Review by jamorham — APPROVED** [review]
Tested and looks good to me

**Comment by jamorham** [comment]
@Apres2707 thank you for your patience and politeness. Testing is successfully completed and this will be merged now. Thank you for your contribution.

**Comment by Navid200** [comment]
@Apres2707 
I would like to announce, to the users, that we have a new device supported by xDrip.  I anticipate to get two sets of questions if I do that.

1- Where can I find out more about this device/product?  Like detail about the product, pros and cons, where it is available, what it can do, ...
2- How do I use this new device with xDrip?  How can I enable this in xDrip?

Any pointers you can give me for how to answer those questions?
Thanks

**Comment by Navid200** [comment]
@Apres2707 Support for this product was added to xDrip.
As a result there may be questions by users as to what it is.
I support users.  If this is a real product, I am willing to provide information to them provided by the product marketing.

I am not going to spend time on this chasing after a product developer if they themselves are not willing, or interested, to do the work to market their own product.

With that information currently lacking, I cannot mention this product to anyone as an option.  Because as soon as I do, they are going to ask me the exact same questions, what is this, where can I find out more about it, how can I buy it, ...
I am absolutely not going to answer those questions by telling them to go chase after someone with an instagram or email address.

Either there is a public web site clearly describing what this product is, and how it can be acquired, or as far as I'm concerned, this product does not exist.

**Comment by Apres2707** [comment]
@Navid200 I understand you and share your opinion. I passed your message on to the product owner. All that I was told that work on the site has begun, but the launch dates could not be identified. In the comments above, it was already reported that at the moment there are no other contacts for obtaining information about the product, unfortunately.
In this regard, the decision on further actions remains with the NightScout team and you, in particular. Previously, it was not obvious that the finished website should be launched by the time the PR was approoved, so this was not given much attention.

I apologize for any inconvenience this may cause. Have a nice day.

**Pull request merged**